### PR TITLE
feat: bump databricks-sdk v0.1.7 and rename RefreshableCredentials to SessionCredentials

### PR DESCRIPTION
**Pull request by pavs23**
This resolves this issue I raised recently https://github.com/databricks/dbt-databricks/issues/348

Root cause is due to the renaming done in this PR: https://github.com/databricks/databricks-sdk-py/pull/116

These types of breaking changes are normal and part of life. Would like to see this get pushed through ASAP though.

### Description
* Bump to the `databricks-sdk` to use the latest version. 
* Rename the `RefreshableCredentials` references to `SessionCredentials`

It would be failing for anyone trying to connect to databricks using `dbt-core` and the `dbt-databricks` adapter if I'm not mistaken.

### Checklist

- [ ] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

**Review by susodapop — CHANGES_REQUESTED** [review]
You need to change the databricks-sdk dependency spec in `setup.py`, else this won't affect the dependency map that `pip` builds during installation.

I had to incorporate these same changes into https://github.com/databricks/dbt-databricks/pull/338 so you can see exactly where the changes are needed. 

Happy to merge your PR instead and rebase my change on yours though 🚀

**Comment by pavs23** [comment]
Hey @susodapop , thanks for the feedback. 
Good spot, I've updated the PR. Ready to merge when you are!

**Review by susodapop — APPROVED** [review]
Thanks! Per the CONTRIBUTING guide, I've updated this PR to hit a staging branch so we can run our e2e tests before merging to `main`.

**Comment by pavs23** [comment]
Thanks @susodapop :)

**Pull request merged**
